src/main/java/goodgenerator/blocks/tileEntity/AntimatterGenerator.java

```diff
@@ -0,0 +1,505 @@
+                .addElement('G', lazy(x -> ofBlock(x.getCoilBlock(1), x.getCoilMeta(1)))) // Annihilation Coil
+                .addElement('B', lazy(x -> ofBlock(x.getCoilBlock(2), x.getCoilMeta(2)))) // Containment Coil
+                .addElement('C', lazy(x -> ofBlock(x.getCasingBlock(2), x.getCasingMeta(2)))) // White Casing
+                .addElement('A', lazy(x -> ofBlock(x.getGlassBlock(), x.getGlassMeta()))) // Glass
+                .addElement('E', lazy(x -> ofBlock(GregTechAPI.sBlockCasings9, 1))) // Filter Casing
+                .addElement(
+                    'H',
+                    lazy(
+                        x -> HatchElementBuilder.<AntimatterGenerator>builder()
+                            .anyOf(HatchElement.ExoticEnergy)
+                            .adder(AntimatterGenerator::addLaserSource)
+                            .casingIndex(x.textureIndex(2))
+                            .dot(1)
+                            .buildAndChain(x.getCasingBlock(2), x.getCasingMeta(2))))
+                .addElement(
+                    'I',
+                    lazy(
+                        x -> buildHatchAdder(AntimatterGenerator.class).atLeast(HatchElement.InputHatch)
+                            .casingIndex(x.textureIndex(1))
+                            .dot(2)
+                            .buildAndChain(x.getCasingBlock(1), x.getCasingMeta(1))))
+                .build();
+        }
+    };
+
+    static {
+        Textures.BlockIcons.setCasingTextureForId(
+            53,
+            TextureFactory.of(
+                TextureFactory.builder()
+                    .addIcon(MACHINE_CASING_ANTIMATTER)
+                    .extFacing()
+                    .build(),
+                TextureFactory.builder()
+                    .addIcon(MACHINE_CASING_ANTIMATTER_GLOW)
+                    .extFacing()
+                    .glow()
+                    .build()));
+    };
+
+    private boolean addLaserSource(IGregTechTileEntity aBaseMetaTileEntity, int aBaseCasingIndex) {
+        IMetaTileEntity aMetaTileEntity = aBaseMetaTileEntity.getMetaTileEntity();
+        if (aMetaTileEntity == null) return false;
+        if (aMetaTileEntity instanceof MTEHatchDynamoTunnel tHatch) {
+            tHatch.updateTexture(aBaseCasingIndex);
+            return mExoticEnergyHatches.add(tHatch);
+        }
+        return false;
+    }
+
+    public AntimatterGenerator(String name) {
+        super(name);
+    }
+
+    public AntimatterGenerator(int id, String name, String nameRegional) {
+        super(id, name, nameRegional);
+    }
+
+    @Override
+    public boolean isCorrectMachinePart(ItemStack aStack) {
+        return true;
+    }
+
+    @Override
+    public RecipeMap<?> getRecipeMap() {
+        return null;
+    }
+
+    @Override
+    public String[] getInfoData() {
+        String[] info = super.getInfoData();
+        info[4] = "Probably makes: " + EnumChatFormatting.RED
```

Comment:
nit: I think this is usually worded as `Probably generates...` in other generators.

```suggestion
        info[4] = "Probably generates: " + EnumChatFormatting.RED
```

Reply:
Need to redo this function entirely anyways, generator works in bursts instead of continously.